org.lflang/src/org/lflang/generator/DockerGeneratorBase.java

```diff
@@ -47,75 +57,88 @@ public class DockerGeneratorBase {
+        public String getFileContent() { return fileContent; }
+        public String getComposeServiceName() { return composeServiceName; }
+        public String getBuildContext() { return buildContext; }
+        public String getContainerName() { return containerName; }
     }
 
+    /**
+     * The interface for data from the code generator.
+     *
+     * Target-specific docker generators can have a class
+     * that implements this interface to specify
+     * what kinds of generator-related data is needed
+     * during docker file generation.
+     */
+    public interface GeneratorData {}
+
     /**
      * The constructor for the base docker file generation class.
      * @param isFederated True if federated execution. False otherwise.
      */
     public DockerGeneratorBase(boolean isFederated) {
-        moduleNameToData = new HashMap<>();
+        dockerDataList = new ArrayList<>();
         composeServices = new StringBuilder();
         this.isFederated = isFederated;
-        nFederates = 0;
     }
 
     /**
-     * Set the `host` of the container
-     * that launches the RTI.
-     * @param host The host to set.
+     * Translate data from the code generator to docker data as
+     * specified in the DockerData class.
+     *
+     * @param generatorData Data from the code generator.
+     * @return docker data as specified in the DockerData class
      */
-    public void setHost(String host) {
-        this.host = host;
-    }
+    abstract protected DockerData generateDockerData(GeneratorData generatorData);
 
     /**
      * Add a federate to the list of federates to generate docker files for.
      *
-     * @param lfModuleName The module name of the federate.
-     * @param federateName The name of the federate's reactor.
-     * @param dockerFilePath The path where the docker file will be written.
-     * @param targetConfig The target config.
-     */
-    public void addFederate(
-        String lfModuleName,
-        String federateName,
-        Path dockerFilePath,
-        TargetConfig targetConfig
-    ) {
-        Map<Key, Object> k = new HashMap<>();
-        k.put(Key.DOCKER_FILE_PATH, dockerFilePath);
-        var dockerFileContent = generateDockerFileContent(lfModuleName);
-        k.put(Key.DOCKER_FILE_CONTENT, dockerFileContent);
-        k.put(Key.DOCKER_COMPOSE_SERVICE_NAME, isFederated ? federateName : lfModuleName.toLowerCase());
-        k.put(Key.DOCKER_BUILD_CONTEXT, isFederated ? federateName : ".");
-        moduleNameToData.put(lfModuleName, k);
-        nFederates++;
-        appendFederateToDockerComposeServices(
-            composeServices,
-            (String) k.get(Key.DOCKER_COMPOSE_SERVICE_NAME),
-            (String) k.get(Key.DOCKER_BUILD_CONTEXT),
-            dockerFilePath.getFileName().toString());
+     * @param generatorData Data from the code generator.
+     */
+    public void addFederate(GeneratorData generatorData) {
```

Comment:
Not sure we need federation lingo here. How about `addProcess` or `addProgram` or `addDeployment` or something in that vain?

Reply:
Or just `addFile`...

Reply:
This is a good point. I will change the name to `addFile`